functions/Test-DbaDatabseBestPractice

```diff
@@ -0,0 +1,305 @@
+	
+	BEGIN
+	{
+		$databases = $psboundparameters.Databases
+		
+		$systemdbs = @("master", "msdb", "model", "tempdb")
+		
+		Function Test-AutoClose
+		{
+			switch ($db.AutoClose)
+			{
+				$true { return $false }
+				default { return $true }
+			}
+		}
+		
+		Function Test-AutoShrink
+		{
+			switch ($db.AutoShrink)
+			{
+				$true { return $false }
+				default { return $true }
+			}
+		}
+		
+		Function Test-PageVerify
+		{
+			if ($server.versionMajor -gt 8)
+			{
+				$bestpractice = "checksum"
+			}
+			else
+			{
+				$bestpractice = "tornpage"
+			}
+    			
+            switch ($db.PageVerify)
+			{
+				$bestpractice { return $true }
+				default { return $false }
+			}
+		}
+		
+		Function Test-AutoGrowth
+		{
+			$dbfiles = @()
+			
+			$filegroups = $db.FileGroups
+			
+			foreach ($filegroup in $filegroups)
+			{
+				foreach ($file in $filegroup.Files) 
+				{
+					$bpgrowthsize = $file.Size * 0.125
+					$isbestpractice = ($file.GrowthType -eq "Percent" -and $file.Growth -ge $bpgrowthsize)
+					
+					switch ($maxsize = $file.MaxSize)
+					{
+						-1 { $maxsize = "Unlimited" }
+					}
+					
+					$dbfiles += New-Object PSObject -Property @{
+						Server = $server.name
+						Database = $db.Name
+						FileGroupName = $filegroup.name
+						FileGroupType = $filegroup.FileGroupType
+						FileSize = $file.Size
+						GrowthType = $file.GrowthType
+						GrowthSize = $file.Growth
+						MaxSize = $maxsize
+						FileType = "Data"
+						IsBestPractice = $isbestpractice
+					}
+				}
+			}
+			
+			foreach ($file in $db.LogFiles)
+			{
+				$bpgrowthsize = $file.Size * 0.125
+				$isbestpractice = ($file.GrowthType -eq "Percent" -and $file.Growth -ge $bpgrowthsize)
+				
+				if ($file.MaxSize -eq -1)
+				{
+					$maxsize = "Unlimited"
+				}
+				else
+				{
+					$maxsize = $file.MaxSize
+					$maxsize = "$maxsize KB"
+				}
+				
+				$dbfiles = New-Object PSObject -Property @{
+					Server = $server.name
+					Database = $db.Name
+					FileGroupName = $file.name
+					FileGroupType = "Not Applicable" 
+					FileSize = $file.Size
+					GrowthType = $file.GrowthType
+					GrowthSize = $file.Growth
+					MaxSize = $maxsize
+					FileType = "Log"
+					IsBestPractice = $isbestpractice
+				}
+			}
+			
+			return $dbfiles
+		}
+
+		$collection = New-Object System.Collections.ArrayList
+	}
+	
+	PROCESS
+	{
+        $processed = New-Object System.Collections.ArrayList
+
+		foreach ($server in $sqlserver) 
+		{
+            if ($server -notin $processed)
+			{
+				$null = $processed.Add($server)
+				Write-Verbose "Connecting to $server"
+			}
+			else
+			{
+				continue
+			}
+
+			Write-Verbose "Attempting to connect to $sqlserver"
+			
+			try
+			{
+				$server = Connect-SqlServer -SqlServer $sqlserver -SqlCredential $SqlCredential
+			}
+			catch
+			{
+				Write-Warning "Can't connect to $sqlserver or access denied. Skipping."
```

Comment:
You don't need this this try/catch because `Connect-SqlServer` includes try/catch. On another note you should provide the exception output `Write-Exception` for try/catch blocks to output the actual reason why it failed or error was caught.